src/js/common/components/form-elements/DateInput.jsx

```diff
@@ -61,6 +61,13 @@ class DateInput extends React.Component {
       daysForSelectedMonth = days[month.value];
     }
 
+    // Our validations require that there is always a "day" field, so
+    // we'll assume the first day of the month even if we're not displaying it.
+    if (this.props.hideDayField) {
+      day.dirty = true;
```

Comment:
This seems like it might eventually cause problems, since we're modifying data from the Redux store that should be immutable. We could have this component ignore the day field and default it to this in the initial veteran object.